### PR TITLE
feat(owlbot): support owlbot:run label

### PR DESCRIPTION
**Pull request by bcoe**
Allow an `owlbot:run` label to be added, similar to `kokoro:run`, which will allow us to run the OwlBot Post-Processor against a fork, or to re-run without pushing (in the case of a failure).

When a label event happens:

1. it will run post-processor if label is `owlbot:run`, and removes it after.
2. it will not run post-processor if it is **not** a fork.
3. it will not run post-processor on fork for labels other than `owlbot:run`, e.g., `cla:yes`.

### CODE REVIEW

File: packages/owl-bot/src/owl-bot.ts [hunk 1]
```diff
@@ -76,15 +78,29 @@ export = (privateKey: string | undefined, app: Probot, db?: Db) => {
...
         logger.info(
           `head ${head.repo.full_name} does not match base ${base.repo.full_name} skipping`
         );
         return;
       }
+      // If the pull request is *not* from a fork, ignore label being added
+      // (we will be running OwlBot post processor for the push itself);
+      if (
+        head.repo.full_name === base.repo.full_name &&
+        context.payload.action === 'labeled'
```
Comment (chingor13) [inline packages/owl-bot/src/owl-bot.ts:97]: Suggestion: if there's separate validation based on the event type, use separate probot `app.on()` handlers and move the common behavior to a helper function.

File: packages/owl-bot/src/owl-bot.ts [hunk 2]
```diff
@@ -76,15 +78,29 @@ export = (privateKey: string | undefined, app: Probot, db?: Db) => {
       if (!installation) {
         throw Error(`no installation token found for ${head.repo.full_name}`);
       }
+      const hasRunLabel = !!context.payload.pull_request.labels.filter(
+        l => l.name === OWLBOT_RUN_LABEL
+      ).length;
```
Comment (chingor13) [inline packages/owl-bot/src/owl-bot.ts:83]: If it's a label event and this label is added, do we want to remove it afterwards (like release-please, kokoro, etc.)

**Review by chingor13 — CHANGES_REQUESTED** [review]

File: packages/owl-bot/src/owl-bot.ts (hunk 2 above)
Comment (bcoe) [reply packages/owl-bot/src/owl-bot.ts:83]: that's a good call, didn't think of that.

File: packages/owl-bot/src/owl-bot.ts [hunk 3]
```diff
@@ -76,15 +78,29 @@ export = (privateKey: string | undefined, app: Probot, db?: Db) => {
...
       // added by a maintainer to trigger the post processor.
-      // TODO(bcoe): add support for "owlbot:run" label.
-      if (head.repo.full_name !== base.repo.full_name) {
+      if (head.repo.full_name !== base.repo.full_name && !hasRunLabel) {
         logger.info(
           `head ${head.repo.full_name} does not match base ${base.repo.full_name} skipping`
         );
         return;
       }
+      // If the pull request is *not* from a fork, ignore label being added
```
Comment (SurferJeffAtGoogle) [inline packages/owl-bot/src/owl-bot.ts:93]: Users may want to re-run Owl Bot if:
1.  There was a transient network or cloud build error.
2.  They don't believe the results of the first run.

File: packages/owl-bot/src/owl-bot.ts [hunk 4]
```diff
@@ -76,15 +78,29 @@ export = (privateKey: string | undefined, app: Probot, db?: Db) => {
       if (!installation) {
         throw Error(`no installation token found for ${head.repo.full_name}`);
       }
+      const hasRunLabel = !!context.payload.pull_request.labels.filter(
+        l => l.name === OWLBOT_RUN_LABEL
+      ).length;
+
       // If the pull request is from a fork, the label "owlbot:run" must be
       // added by a maintainer to trigger the post processor.
```
Comment (SurferJeffAtGoogle) [inline packages/owl-bot/src/owl-bot.ts:86]: Which line of code confirms that the label was added by a maintainer?

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]

File: packages/owl-bot/src/owl-bot.ts (hunk 4 above)
Comment (chingor13) [reply packages/owl-bot/src/owl-bot.ts:86]: adding a label requires >= "Triage" permissions

File: packages/owl-bot/src/owl-bot.ts (hunk 1 above)
Comment (bcoe) [reply packages/owl-bot/src/owl-bot.ts]: I've split into two `on`s, I think it simplifies logic.

File: packages/owl-bot/src/owl-bot.ts (hunk 3 above)
Comment (bcoe) [reply packages/owl-bot/src/owl-bot.ts]: Labeling now works when *not* from fork.

File: packages/owl-bot/src/owl-bot.ts [hunk 5]
```diff
@@ -63,105 +63,167 @@ export = (privateKey: string | undefined, app: Probot, db?: Db) => {
...
+
+    if (!installation) {
+      throw Error(`no installation token found for ${head}`);
+    }
+
+    const hasRunLabel = !!context.payload.pull_request.labels.filter(
+      l => l.name === OWLBOT_RUN_LABEL
+    ).length;
+
+    // If the pull request is *not* from a fork, ignore label being added
```
Comment (SurferJeffAtGoogle) [inline packages/owl-bot/src/owl-bot.ts:82]: Is this comment still accurate?

**Review by chingor13 — APPROVED** [review]

File: packages/owl-bot/src/owl-bot.ts (hunk 5 above)
Comment (bcoe) [reply packages/owl-bot/src/owl-bot.ts]: Removed stale comment.

**Review by SurferJeffAtGoogle — APPROVED** [review]

**Pull request merged**